### PR TITLE
Remove `-alpha` tags from Airbyte release versions

### LINKED ISSUE #14022: Remove alpha tags from OSS Releases
We want to remove the `-alpha` from our release version for the OSS Platform.

There are a few places to change.

* check actions and scripts for assumptions on the -alpha name (regex?) 
* be sure to sure to actually do a release and see that things work with the new name
* do a cloud release too and ensure the non-alpha release is OK
* ensure that docs don't mention `-alpha`

### PR DESCRIPTION
**Pull request by evantahler**
Closes https://github.com/airbytehq/airbyte/issues/14022

This PR removes `-alpha` from the Airbyte versions and releases.  As we are using [Semver and on a pre-v1](https://semver.org/#spec-item-4), we are already broadcasting that we have an unstable (read: rapidly changing) product and API.  Airbyte has thousands of companies using the product already, both on-prem and via Airbyte cloud, so the `-alpha` label is no longer appropriate.   

We use the [`bumpversion`](https://pypi.org/project/bumpversion/) tool to bump the version numbers in various files via the `release_version` action, running the `bump_version` script. This PR updates the bumpversion config file and removes the trailing `-alpha` tags from versioned files.  

Note: Once this PR is merged, we will be able to release another OSS patch release, which should update all these files to `0.40.1` (next version) (no `-alpha`)

This PR is hard to test, but you can check out this branch locally and then run `bumpversion patch` and see the changes:

```diff
diff --git a/.bumpversion.cfg b/.bumpversion.cfg
index 5ffda3c5a0..decd983c4b 100644
--- a/.bumpversion.cfg
+++ b/.bumpversion.cfg
@@ -1,9 +1,9 @@
 [bumpversion]
-current_version = 0.40.0
+current_version = 0.40.1
 commit = False
 tag = False
 parse = (?P<major>\d+)\.(?P<minor>\d+)\.(?P<patch>\d+)(\-[a-z]+)?

 [bumpversion:file:.bumpversion.cfg]
diff --git a/.env b/.env
index 9e1e7b3897..bb69ee03f8 100644
--- a/.env
+++ b/.env
@@ -10,7 +10,7 @@


 ### SHARED ###
-VERSION=0.40.0
+VERSION=0.40.1

 # When using the airbyte-db via default docker image
 CONFIG_ROOT=/data
diff --git a/airbyte-bootloader/Dockerfile b/airbyte-bootloader/Dockerfile
index 149c829e4c..299372b22b 100644
--- a/airbyte-bootloader/Dockerfile
+++ b/airbyte-bootloader/Dockerfile
@@ -2,7 +2,7 @@ ARG JDK_VERSION=19-slim-bullseye
 ARG JDK_IMAGE=openjdk:${JDK_VERSION}
 FROM ${JDK_IMAGE}

-ARG VERSION=0.40.0
+ARG VERSION=0.40.1

 ENV APPLICATION airbyte-bootloader
 ENV VERSION ${VERSION}
diff --git a/airbyte-container-orchestrator/Dockerfile b/airbyte-container-orchestrator/Dockerfile
index 32558a97f5..1196cb40b0 100644
--- a/airbyte-container-orchestrator/Dockerfile
+++ b/airbyte-container-orchestrator/Dockerfile
@@ -28,7 +28,7 @@ RUN echo "deb [signed-by=/usr/share/keyrings/kubernetes-archive-keyring.gpg] htt
 RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y kubectl

 # Don't change this manually.  Bump version expects to make moves based on this string
-ARG VERSION=0.40.0
+ARG VERSION=0.40.1

 ENV APPLICATION airbyte-container-orchestrator
 ENV VERSION=${VERSION}
diff --git a/airbyte-metrics/reporter/Dockerfile b/airbyte-metrics/reporter/Dockerfile
index de12c75045..9a2713c7df 100644
--- a/airbyte-metrics/reporter/Dockerfile
+++ b/airbyte-metrics/reporter/Dockerfile
@@ -2,7 +2,7 @@ ARG JDK_VERSION=19-slim-bullseye
 ARG JDK_IMAGE=openjdk:${JDK_VERSION}
 FROM ${JDK_IMAGE} AS metrics-reporter

-ARG VERSION=0.40.0
+ARG VERSION=0.40.1

 ENV APPLICATION airbyte-metrics-reporter
 ENV VERSION ${VERSION}
diff --git a/airbyte-server/Dockerfile b/airbyte-server/Dockerfile
index 14bfdaf4f9..78ae878784 100644
--- a/airbyte-server/Dockerfile
+++ b/airbyte-server/Dockerfile
@@ -4,7 +4,7 @@ FROM ${JDK_IMAGE} AS server

 EXPOSE 8000

-ARG VERSION=0.40.0
+ARG VERSION=0.40.1
```

### CODE REVIEW

**Comment by evantahler** [comment]
It is expected that the [Ensure all required Docker images exist on Dockerhub](https://github.com/airbytehq/airbyte/runs/7905158142?check_suite_focus=true#logs) action fail until we release the next version of OSS... which we should do immediately after this PR is merged.

**Comment by evantahler** [comment]
@davinchia I *think* that the OSS version names don't matter too much for cloud deployments any more (https://github.com/airbytehq/airbyte-cloud/pull/2332/) as the SHA is what is used now.  Can you please confirm that the changes here will be OK for cloud?

**Comment by davinchia** [comment]
@evantahler ack. will take a look later today

File: charts/airbyte/Chart.yaml [hunk 1]
```diff
@@ -61,4 +60,4 @@ dependencies:
   - condition: pod-sweeper.enabled
     name: pod-sweeper
     repository: "https://airbytehq.github.io/helm-charts/"
-    version: placeholder
\ No newline at end of file
+    version: placeholder
```
Comment (supertopher) [inline charts/airbyte/Chart.yaml:63]: ❤️

**Review by supertopher — COMMENTED** [review]
this looks like a good change to me.  
I think removing alpha makes sense for external communication.

I added Peter who has context on bump version in infra

**Review by davinchia — APPROVED** [review]
That looks right to me! 

My understanding is also that Cloud has been updated to take refs, so we should be good now.

Honestly, it's tough to test these things so.. I would just go for it. I'm available tomorrow to pair on ironing out kinks (if any).

**Review by git-phu — APPROVED** [review]
looks good!

Also, cloud references oss based on commit sha or git tags, and in the case of tags it doesn't assume anything about the structure/format of the tag name other than that the tag starts with a `v`, so this change should be fully compatible with cloud

**Pull request merged**